src/main/java/ch/naviqore/raptor/Raptor.java

```diff
@@ -473,16 +476,25 @@ private static void validateArrivalHandicap(int arrivalHandicap) {
             }
         }
 
-        private int[] validateAndGetStopIdx(Collection<String> stopIds) {
-            return stopIds.stream().mapToInt(this::validateAndGetStopIdx).toArray();
-        }
+        private Map<Integer, Integer> validateStops(Map<String, Integer> stops) {
+            if (stops.isEmpty()) {
+                throw new IllegalArgumentException("At least one stop ID must be provided.");
+            }
+            // Loop over all stop pairs
+            Map<Integer, Integer> validStopIds = new HashMap<>();
+            for (Map.Entry<String, Integer> entry : stops.entrySet()) {
+                if (stopsToIdx.containsKey(entry.getKey())) {
```

Comment:
Can you unpack the entry? 
String stopId = entry.getValue()
... = entry.getValue()

Just for readability, otherwise i will not get the content of the map in a week🤣

And a further wish: Is there a more concise name than handicap time? Something like targetWalkingDuration, lastMileDuration, ... although this is not really beautiful, but better understandable what it means (at least for me).

Reply:
Unpacks addressed in 94f7d96
Handicap time rename in c1d67a8

Reply:
Thanks!